tests/jvm/src/test/scala/scala/meta/tests/semanticdb/PreciseTypeSuite.scala

```diff
@@ -0,0 +1,78 @@
+package scala.meta.tests.semanticdb
+
+import scala.meta.internal.semanticdb.scalac.SymbolMode
+import scala.meta.internal.semanticdb3.Print
+import scala.meta.internal.{semanticdb3 => s}
+
+class PreciseTypeSuite extends SemanticdbSuite(symbols = SymbolMode.AllWithMaterializedTypes) {
+
+  private def testActualTypes(code: String)(occurencesTypes: String) = test(code){
+    val db = computeDatabaseFromSnippet(code)
+    val allOccurences = db.occurrences ++ db.synthetics.flatMap(_.text).flatMap(_.occurrences)
+
+    def printOccurence(occ: s.SymbolOccurrence) = occ.materializedTpe.map( tpe =>
+          s"${Print.range(occ.range.get)} ${Print.tpe(db, tpe)}"
+      )
+
+    val occurencesStrings = for {
+      syn <- db.synthetics
+      text = syn.text.get
+      occ <- text.occurrences
+    } yield printOccurence(occ).map(t => s"${Print.range(syn.range.get)} ~> $t")
+
+    val typeLines = db.occurrences.map(printOccurence) ++ occurencesStrings
+    val typeString = typeLines.flatten.sorted.mkString("\n")
+    assert(typeString == occurencesTypes)
+  }
+
+  testActualTypes(
+    "object L2 { val head = Array(1, 2, 3)(0) }"
+  )(
+    """[0:23..0:28) scala.Array.
+      |[0:28..0:28) ~> [0:2..0:7) scala.Array.apply(Int,Int*).(x)scala.Array.apply(Int,Int*).(xs)scala.Array#scala.Int#
```

Comment:
I'm not sure what `scala.Array.apply(Int,Int*).(x)scala.Array.apply(Int,Int*).(xs)scala.Array#scala.Int#` means. This looks like multiple concatenated types, but I don't quite get why there are several of them. Could you elaborate?